src/generator/backend/generator.js

```diff
@@ -0,0 +1,253 @@
+        caption: track.date,
+        tracks: []
+      })
+    }
+    dateMap.get(track.date).tracks.push(track);
+  })
+
+  let dates = Array.from(dateMap.values())
+  dates.forEach((date) => date.tracks.sort(byProperty('sortKey')))
+
+  return dates;
+}
+
+function byProperty(key) {
+  return (a, b) => {
+    if (a[key] > b[key]) {
+      return 1;
+    }
+    if (a[key] < b[key]) {
+      return -1;
+    } else {
+      return 0;
+    }
+  }
+}
+
+function zeroFill(num) {
+  if (num === undefined) {
+    return ""
+  }
+  if (num >= 10) {
+    return num.toString();
+  } else {
+    return '0' + num.toString();
+  }
+}
+
+function foldByTrack(sessions, speakers) {
+  let trackData = new Map()
+  let speakersMap = new Map(speakers.map((s) => [s.id, s]));
+
+  sessions.forEach((session) => {
+    if (!session.start_time) {
+      return
+    }
+
+    // generate slug/key for session
+    let date = moment(session.start_time).format('YYYY-MM-DD');
+    let slug = date + '-' + slugify(session.track.name);
+    let track = null;
+
+    // set up track if it does not exist
+    if (!trackData.has(slug)) {
+      track = {
+        title: session.track.name,
+        date: moment(session.start_time).locale('de').format('ddd D. MMM') + ' / ' + moment(session.start_time).format('ddd, Do MMM'),
+        slug: slug,
+        sortKey: date + '-' + zeroFill(session.track.order),
+        sessions: []
+      }
+      trackData.set(slug, track);
+    } else {
+      track = trackData.get(slug);
+    }
+
+    track.sessions.push({
+      start: moment(session.start_time).utcOffset(2).format('HH:mm'),
+      title: session.title,
+      type: session.type,
+      location: session.location,
+      speakers: session.speakers.map(speakerNameWithOrg).join(', '),
+      speakers_list: session.speakers.map((speaker) => speakersMap.get(speaker.id)),
+      description: session.description,
+      session_id: session.session_id,
+      sign_up: session.sign_up,
+      video: session.video,
+      slides: session.slides,
+      audio: session.audio
+    })
+  });
+
+  let tracks = Array.from(trackData.values())
+  tracks.sort(byProperty('sortKey'))
+
+  return tracks;
+}
+
+function getCopyrightData(services) {
+  let copyright = services.copyright
+  return copyright;
+}
+
+function foldByLevel(sponsors) {
+  let levelData = {}
+
+  sponsors.forEach((sponsor) => {
+    if (levelData[sponsor.level] === undefined) {
+      levelData[sponsor.level] = []
+    }
+
+    let sponsorItem = {
+      divclass: '',
+      imgsize: '',
+      name: sponsor.name,
+      image: sponsor.image,
+      link: sponsor.link,
+      level: sponsor.level,
+      description: sponsor.description,
+      type: sponsor.type
+    }
+
+    switch (sponsorItem.level) {
+      case '1':
+        sponsorItem.divclass = "largeoffset col-md-4"
+        sponsorItem.imgsize = "large"
+        break;
+      case '2':
+      default:
+        sponsorItem.divclass = "mediumoffset col-md-2"
```

Comment:
![Codacy](https://www.codacy.com/assets/images/favicon.png) Issue found: [Mixed double and single quotes.](https://www.codacy.com/app/triplez/open-event-webapp/file/2708617232/issues/source?bid=3233857&fileBranchId=3342834#l170)